src/schema.ts

```diff
@@ -198,6 +198,35 @@ export class Schema {
     }
   }
 
+  /**
+   * Given an EncodingQuery with a timeUnit, returns true if the number of
+   * dates that are completely distinct in all parts of the timeUnit is greater than 1.
+   * i.e.
+   * ('yearmonth', [Jan 1 2000, Feb 2 2000] returns false)
+   * ('yearmonth', [Jan 1 2000, Feb 2 2001] returns true)
+   */
+  public timeUnitHasVariation(encQ: EncodingQuery): boolean {
+    // if there is no variation in `date`, there should not be variation in `day`
+    if (encQ.timeUnit === TimeUnit.DAY) {
+      const dateEncQ: EncodingQuery = extend({}, encQ);
+      dateEncQ.timeUnit = TimeUnit.DATE;
+      if (this.cardinality(dateEncQ, false, true) <= 1) {
+        return false;
+      }
+    }
+
+    let fullTimeUnit = encQ.timeUnit;
+    for (let singleUnit of SINGLE_TIMEUNITS) {
+      if (containsTimeUnit(fullTimeUnit as TimeUnit, singleUnit)) {
```

Comment:
This is slightly inefficient as we have to always loop through all of them.  Maybe file an issue so that we can refactor all occurrence of `containsTimeUnit`?  (No need to do now.)